src/pages/ProfilesExplorerView/components/SceneProfilesExplorer/SceneProfilesExplorer.tsx

```diff
@@ -266,13 +266,33 @@ export class SceneProfilesExplorer extends SceneObjectBase<SceneProfilesExplorer
     sceneGraph.findByKeyAndType(this, 'groupBy', GroupByVariable).changeValueTo(GroupByVariable.DEFAULT_VALUE);
     sceneGraph.findByKeyAndType(this, 'panel-type-switcher', ScenePanelTypeSwitcher).reset();
 
-    // preserve existing filters only when switching to "Labels" or "Flame graph"
-    if (![ExplorationType.LABELS, ExplorationType.FLAME_GRAPH].includes(nextExplorationType as ExplorationType)) {
+    // preserve existing filters only when switching to "Labels", "Flame graph" or "Diff flamge graph"
+    if (
```

Comment:
Definitely a great improvement 🙌  

One small nit: I was a bit surprised with following flow:

* I set some labels in Labels view
* I go to Diff View and remove (or change both values to something else)
* I switch back to Labels view (the labels are restored)

If I remove all labels in Diff View and go to labels I kind of expected to have no labels. I'm not sure how it should work in general though 😅  There are multiple cases (one label changed or removed), so it's not easy to make it super clear all the time. An alternative would be to have another input with "common labels" applied to both sides of the view + side filters added to add more granular labels. But that could add more noise to the UI 🤷 

Not a blocker, just food for thought. Something we can keep an eye on and double check with UX.

Reply:
Thank you for the feedback. It's a tricky one indeed for the cases you mentioned. Let's see with UX.